CIP Cardano Multichain Router b02eb0e3a2bf408a85bdbe6a4ce5603b.md

```diff
@@ -0,0 +1,395 @@
+
+## Router
+
+Cardano will be integrated with Multichain router V3, which was lauched on June 4th 2021. The Router can allow assets to be transferred among all others connected chains including Cardano. Check more about router here:
+
+[https://docs.multichain.org/getting-started/how-it-works/cross-chain-router](https://docs.multichain.org/getting-started/how-it-works/cross-chain-router)
+
+# User Cases
+
+Tokens on different chains can be bridged to Cardano, but let’s simplify it to the scenario between Ethereum and Cardano.
+
+On Cardano side, the Multichan SMPC nodes manage a MPC address, which locks the Cardano native tokens bridged to Ethereum and mint the wrapped tokens bridged from Ethereum.
+
+On Ethereum side, the Multichain SMPC nodes will call Multichain router smart contract to send the Ethereum native token in the pool to users and mint the wrapped tokens bridged from Cardano. 
+
+## A. ****Bridging Ethereum native tokens from Ethereum to Cardano****
+
+1, User send USDC to the USDC pool on Ethereum and indicate the receiving address on Cardano.
+
+2, The SMPC nodes verify the user transaction on Ethereum independently.
+
+3, After the verification, the SMPC nodes work together to sign a minting transaction on Cardano.
+
+4, The SMPC nodes mint wrapped USDC to user’s address on Cardano. 
+
+## B. Bridging wrapped Ethereum native token from Cardano to Ethereum
+
+1, User send wrapped USDC to the SMPC address on Cardano and indicate the receiving address on Ethereum.
+
+2, The SMPC nodes verify the user transacton on Cardano independently.
+
+3, After the verification, the SMPC nodes work together to sign a transfer transaction on Ethereum.
+
+4, The SMPC nodes send the native USDC to user’s address on Ethereum. 
+
+## C. Bridging Cardano native token from Cardano to Ethereum
+
+1, User send native ADA to the SMPC address on Cardano and indicate the receiving address on Ethereum.
+
+2, The  SMPC nodes verify the user transaction on Cardano independently.
+
+3, After the verification, the SMPC nodes work together to sign a minting transaction on Ethereum.
+
+4, The SMPC nodes mint wrapped ADA to user’s address on Ethereum.
+
+## D. Bridging wrapped Cardano native token from Ethereum to Cardano
+
+1, User send wrapped ADA to the wrapped ADA pool on Ethereum and indicate the receiving address on Cardano.
+
+2, The SMPC nodes verify the user transaction on Ethereum independently.
+
+3, After the verification, the SMPC nodes work together to sign a transfer transaction on Cardano.
+
+4, The SMPC nodes send the native ADA to user’s address on Cardano.
+
+# Specification
+
+## Major Modules
+
+### SwapOut transaction
+
+Cardano: Transfer to SMPC address with metadata indicating key/value:
+
+```json
+{
+    "metadata": {
+        "key": "123",
```

Comment:
Since the metadata is tied to a transaction, not an UTxO, there are additional requirements that need to be defined.
1. User includes multiple UTxO for the bridge address with different tokens in the same tx (e.g. they could send 2 different utxos, one containing 100 wUSDC and the second 1 wBTC)
2. user includes multiple UTxOs for the bridge address with the same token (2 utxos, 100 wUSDC and 250 wUSDC, in total 350 wUSDC)
3. user includes 1 UTxOs with 2 whitelisted assets: (100 wUSDC + 1 wBTC + for good measure 500 ADA)

Reply:
Response to a similar query by Luoxi. It'd be good to include this in the document:

> > 3. what if there are multiple tokens?
> > 4. what amount is considered here? Additional ada amount? what are the requirements?
> > 5. What’s is a legitimate range and who defines it?

> 3, Multiple tokens in one transaction is supported. Every token is treaded as a separate bridging transaction. txHash+index in the tokenList is one bridging transaction.
> 4, The amount of bridged token is considered here.
> 5, The legitimate range is between the minimum and maximum amount specified in the config contract, which is updated in the token deployment process.